### ISSUE TITLE
Review enum-related behavior

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#935]
As of https://github.com/fractal-analytics-platform/fractal-server/pull/934#issuecomment-1782842865, we may be fixing the CI with sqlmodel >0.0.8, but we still need to make sure that the following test works.

* Add a migration that changes job.status type into an actual enum, as it would be autogenerated with sqlmodel 0.0.9
* Run fractal-server with postgres (and gunicorn, if possible)
* Look for unexpected behavior, especially by running jobs (because it's their status that could be broken)

**Comment by tcompa** [comment]
Note: due to the bug in sqlmodel<=0.0.8, that was representing enums as autostrings, we have no enums in our migrations.
Therefore this specific error should be impossible until we update sqlmodel to >=0.0.9.

**Comment by tcompa** [comment]
Ref https://github.com/tiangolo/sqlmodel/issues/164

**Comment by mfranzon** [comment]
As long as SQLITE does not support [ENUM type](https://www.sqlite.org/datatype3.html), do we want to make an `if` branch to store `enum` in postgres and string in sqlite?

**Comment by tcompa** [comment]
Where would you place the `if`? In the migration scripts?

**Comment by mfranzon** [comment]
> Where would you place the `if`? In the migration scripts?

Yes, I am thinking the same but honestly I don't like too much this approach

**Comment by tcompa** [comment]
We re-discussed this with @mfranzon, and had a look at known issues like:
* https://github.com/sqlalchemy/alembic/issues/278

- [ ] We think it's not worth it to have actual enums in the db, for the moment. We'll keep them for all serializer (that is, Pydantic models), but not for the actual tables (that is, `SQLModel` models).
- [ ] Let's add a Pydantic validator to the SQLModel objects, that checks e.g. that `ApplyWorkflow.status` complies with `JobStatusType`.

In this way we don't risk breaking changes with due to the recent sqlmodel update.

We should also keep it in mind when adding new columns in the future.
